analyzers/src/SonarAnalyzer.CSharp/Trackers/CSharpObjectInitializationTracker.cs

```diff
@@ -212,10 +212,13 @@ private bool IsLaterAssignedWithAllowedValue(IObjectCreation objectCreation, Sem
                     .OfType<AssignmentExpressionSyntax>()
                     .Any(TrackedPropertySetWithAllowedValue);
 
-            bool TrackedPropertySetWithAllowedValue(AssignmentExpressionSyntax assignment) =>
-                variableSymbol.Equals(GetAssignedVariableSymbol(assignment, semanticModel))
-                && IsTrackedPropertyName(assignment.Left)
-                && IsAllowedValue(assignment.Right, semanticModel);
+            bool TrackedPropertySetWithAllowedValue(AssignmentExpressionSyntax assignment)
+            {
+                var assignmentMap = assignment.MapAssignmentArguments();
+                return assignmentMap.Any(x => variableSymbol.Equals(GetAssignedVariableSymbol(x.Left, semanticModel))
+                                              && IsTrackedPropertyName(x.Left)
```

Comment:
`IsTrackedPropertyName` should go first, because it is not querying the semantic model.